src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -86,12 +126,77 @@ mf::XWaylandSurface::XWaylandSurface(
       seat(seat),
       shell{shell},
       window(event->window),
-      props_dirty(true),
-      init{
-          event->parent,
-          {event->x, event->y},
-          {event->width, event->height},
-          (bool)event->override_redirect}
+      property_handlers{
+          property_handler<std::string const&>(
+              connection,
+              window,
+              XCB_ATOM_WM_CLASS,
+              [this](auto value)
+              {
+                  std::lock_guard<std::mutex> lock{mutex};
+                  this->pending_spec(lock).application_id = value;
+              }),
+          property_handler<std::string const&>(
+              connection,
+              window,
+              XCB_ATOM_WM_NAME,
+              [this](auto value)
+              {
+                  std::lock_guard<std::mutex> lock{mutex};
+                  this->pending_spec(lock).name = value;
+              }),
+          property_handler<std::string const&>(
+              connection,
+              window,
+              connection->net_wm_name,
+              [this](auto value)
+              {
+                  std::lock_guard<std::mutex> lock{mutex};
+                  this->pending_spec(lock).name = value;
+              }),
+          property_handler<xcb_window_t>(
+              connection,
+              window,
+              connection->wm_transient_for,
+              [this](xcb_window_t value)
+              {
+                  std::shared_ptr<scene::Surface> parent_scene_surface;
+
+                  auto const parent_surface = this->xwm->get_wm_surface(value);
+                  if (parent_surface)
+                  {
+                      std::lock_guard<std::mutex> parent_lock{parent_surface.value()->mutex};
+                      parent_scene_surface = parent_surface.value()->weak_scene_surface.lock();
+                  }
+
+                  {
+                      std::lock_guard<std::mutex> lock{this->mutex};
+                      auto const parent = parent_surface.value()->weak_scene_surface.lock(); // Can be nullptr
```

Comment:
This looks odd: We've a test `if (parent_surface)` followed (regardless of the outcome) by using `parent_surface.value()`. If `parent_surface` needs testing in the first case, how can we use it in the second?

Reply:
Yikes, good catch